### ISSUE TITLE
Ledger Identity Verification ("raw") signing

### ISSUE THREAD

**Issue by carlosala** [polkadot-js/common#1848]
* **I'm submitting a ...**

  - [ ] Bug report
  - [x] Feature request
  - [ ] Support request
  - [ ] Other

* **What is the current behavior and expected behavior?**

We received at Zondax feedback from users asking to implement Raw data signing. We'd like to sync with polkadot-js the release of this feature!

Which would be the best option for you to do this? We were thinking on adding a helper function in https://github.com/Zondax/ledger-substrate-js so you only have to call there passing the derivation path.
As far as we understood, blobs are only going to be `<Bytes>payload</Bytes>`, so we'd only sign blobs like those, showing on-screen the payload if it's printable, otherwise showing the hex representation of it.

**Comment by jacogr** [comment]
That sounds spot-on.

**Comment by carlosala** [comment]
@jacogr I am starting to develop everything. Could you create a POC for it in the apps (or extension)? You have `@zondax/ledger-substrate` version `0.41.0.beta-1` that adds the method `signRaw`. That should be the method to call when signing raw data.
You have the code [here](https://github.com/Zondax/ledger-substrate-js/pull/92/commits/24c127e073f5c00492a7759c250951a7ab947c0b).
It should change nothing from your side besides that.
Thanks!

**Comment by jacogr** [comment]
There is no "quick test in apps". It basically means apps first needs to be updated to the latest common version that has support for it.

- common repo - Add support with an additional [signRaw method](https://github.com/polkadot-js/common/blob/49afd1fe0906f3ba22d79e05c8685a56af60fad4/packages/hw-ledger/src/bundle.ts#L91-L103)
- apps repo - For Ledger wrap using `useLedger()` for [any hardware device](https://github.com/polkadot-js/apps/blob/dddee2eb70a4ac914b8c4b654e84b8be37380bfa/packages/page-signing/src/Sign.tsx#L64-L72) - will need some tweaking to provide a `Signer`, possibly [extending the existing with signRaw](https://github.com/polkadot-js/apps/blob/dddee2eb70a4ac914b8c4b654e84b8be37380bfa/packages/react-signer/src/signers/LedgerSigner.ts#L23-L28)
- extension repo - can be looked at down the line when apps is there, should follow something similar with [extension to the Signers](https://github.com/polkadot-js/extension/blob/4b6b53d2f9cdd466dc6a4f0ab2b7d673ec976d41/packages/extension-ui/src/Popup/Signing/LedgerSign.tsx#L51-L57)

**Comment by carlosala** [comment]
Ok, then I can merge and release stable `0.41.0` so you can proceed.

**Comment by carlosala** [comment]
@jacogr I released `v0.41.0`. Could you please tell me an approximate ETA? Just for having a local apps running with that enabled. There's a new runtine upgrade coming and we'd like to ship this feature with it after testing it 👌🏻

**Comment by carlosala** [comment]
I made a small mistake in v0.41.0, I released v0.41.1 fixing it.

**Comment by carlosala** [comment]
Hi @jacogr!
Not sure which is the progress of this, but from our side is 100% done! Latest Polkadot and Kusama updates (v9430) already have this feature implemented.

**Comment by jacogr** [comment]
Yes, I'm the blocker - I don't have much time at this point due to health/hospital/tests/etc.

Having said that, there has been a common release yesterday, so the new stuff is out in the wild. It will obviously take "some time" at this point before I may have time to add functionality for that in any other repos such as apps or extension. (API release using the latest `@polkadot/util` will probably come sooner, but certainly not days away). 

TL;DR Closing. Functionality as added is released in the latest https://github.com/polkadot-js/common/releases/tag/v12.4.1

EDIT: It probably would make sense to log the enhancements for other polkadot-js projects if this is to be added so it gets in the queue and can be picked up by anybody. (The issues here assumes that "all will be done everywhere", but rather the focus here is only on adding the core functionality and exposing that in the utils)
